sonic-pcied/scripts/pcied

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python2
+
+"""
+    pcied
+    PCIe device monitoring daemon for SONiC
+"""
+
+try:
+    import os
+    import sys
+    import signal
+    import threading
+    import subprocess
+    from sonic_daemon_base.daemon_base import Logger
+    from sonic_daemon_base.daemon_base import DaemonBase
+    import swsssdk
+except ImportError, e:
+    raise ImportError(str(e) + " - required module not found")
+
+#
+# Constants ====================================================================
+#
+SYSLOG_IDENTIFIER = "pcied"
+
+PCIE_RESULT_REGIX = "PCIe Device Checking All Test"
+PCIE_TABLE_NAME = "PCIE_STATUS"
+
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+PCIE_CONF_FILE = 'pcie.yaml'
+SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+HWSKU_KEY = 'DEVICE_METADATA.localhost.hwsku'
+PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
+
+PCIED_MAIN_THREAD_SLEEP_SECS = 60
+REDIS_HOSTIP = "127.0.0.1"
+
+# Global logger class instance
+logger = Logger(SYSLOG_IDENTIFIER)
+
+#
+# Helper Function ==============================================================
+#
+
+
+def get_platform_and_hwsku():
+    try:
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-H', '-v', PLATFORM_KEY],
+                                stdout=subprocess.PIPE,
+                                shell=False,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        platform = stdout.rstrip('\n')
+
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-d', '-v', HWSKU_KEY],
+                                stdout=subprocess.PIPE,
+                                shell=False,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        hwsku = stdout.rstrip('\n')
+    except OSError, e:
+        raise OSError("Cannot detect platform")
+
+    return (platform, hwsku)
+
+#
+# Daemon =======================================================================
+#
+
+
+class DaemonPcied(DaemonBase):
+    def __init__(self):
+        DaemonBase.__init__(self)
+
+        self.timeout = PCIED_MAIN_THREAD_SLEEP_SECS
+        self.stop_event = threading.Event()
+
+        self.state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+        self.state_db.connect("STATE_DB")
+
+    def check_pcie_devices(self):
+        platform, hwsku = get_platform_and_hwsku()
+        pciefilePath = "/".join([PLATFORM_ROOT_PATH, platform, "plugins", PCIE_CONF_FILE])
+        sys.path.append(os.path.abspath(pciefilePath))
+        if not os.path.exists(pciefilePath):
+            logger.log_error("Platform pcie configuration file doesn't exist! exit pcied")
+            sys.exit("Platform PCIe Configuration file doesn't exist!")
```

Comment:
I would suggest moving this check outside this function and do it before entering the loop, or it will periodically print out error msg on the platform which doesn't have this supported?